tests/tests_geomstats/test_geometry/test_discrete_curves.py

```diff
@@ -2,224 +2,191 @@
+        equip=False,
+    ).equip_with_metric(L2CurvesMetric)
 
-@pytest.fixture(
-    scope="class",
-    params=[
-        (2, random.randint(5, 10)),
-        (3, random.randint(5, 10)),
-    ],
-)
-def shape_bundles(request):
-    dim, k_sampling_points = request.param
+    testing_data = L2CurvesMetricTestData()
 
-    ambient_manifold = Euclidean(dim=dim)
-    space = DiscreteCurves(
-        ambient_manifold, k_sampling_points=k_sampling_points, equip=True
-    )
-    request.cls.total_space = request.cls.base = space
 
-    request.cls.bundle = SRVShapeBundle(space)
+class TestElasticTranslationMetric(
+    PullbackDiffeoMetricTestCase, metaclass=DataBasedParametrizer
+):
+    _k_sampling_points = random.randint(5, 10)
+    _a = gs.random.uniform(low=0.5, high=2.5, size=1)
+
+    space = DiscreteCurvesStartingAtOrigin(
+        ambient_dim=2, k_sampling_points=_k_sampling_points, equip=False
+    ).equip_with_metric(ElasticTranslationMetric, a=_a)
 
-    request.cls.sphere = Hypersphere(dim=dim - 1)
+    testing_data = ElasticTranslationMetricTestData()
 
 
-@pytest.mark.usefixtures("shape_bundles")
-class TestSRVShapeBundle(SRVShapeBundleTestCase, metaclass=DataBasedParametrizer):
-    testing_data = SRVShapeBundleTestData()
+class TestSRVTranslationMetric(
+    PullbackDiffeoMetricTestCase, metaclass=DataBasedParametrizer
+):
+    _ambient_dim = random.randint(2, 3)
+    _k_sampling_points = random.randint(5, 10)
 
+    space = DiscreteCurvesStartingAtOrigin(
+        ambient_dim=_ambient_dim,
+        k_sampling_points=_k_sampling_points,
+    )
+    testing_data = SRVTranslationMetricTestData()
 
-@pytest.fixture(
-    scope="class",
-    params=[
-        (2, random.randint(5, 10)),
-        (3, random.randint(5, 10)),
-    ],
-)
-def spaces_with_quotient(request):
-    dim, k_sampling_points = request.param
 
-    ambient_manifold = Euclidean(dim=dim)
-    space = DiscreteCurves(
-        ambient_manifold, k_sampling_points=k_sampling_points, equip=True
+class TestSRVTranslationReparametrizationBundle(
+    SRVTranslationReparametrizationBundleTestCase, metaclass=DataBasedParametrizer
+):
+    _ambient_dim = random.randint(2, 3)
+    _k_sampling_points = random.randint(5, 10)
+
+    total_space = base = DiscreteCurvesStartingAtOrigin(
+        ambient_dim=_ambient_dim,
+        k_sampling_points=_k_sampling_points,
     )
+    bundle = SRVTranslationReparametrizationBundle(total_space)
+
+    data_generator = base_data_generator = ShapeBundleRandomDataGenerator(total_space)
+    testing_data = SRVTranslationReparametrizationBundleTestData()
+
+
+@pytest.mark.smoke
+class TestAlignerCmp(TestCase, metaclass=DataBasedParametrizer):
+    total_space = DiscreteCurvesStartingAtOrigin(k_sampling_points=10)
+    bundle = SRVTranslationReparametrizationBundle(total_space)
 
-    space.equip_with_group_action("reparametrizations")
-    space.equip_with_quotient_structure()
+    aligner = IterativeHorizontalGeodesic()
```

Comment:
NIT: this class' name should probably end with Aligner, to match the logic of line 178.

Reply:
Done.